hr_leave_dashboard_extra_hours/models/hr_leave_type.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2023 Janik von Rotz <janik.vonrotz@mint-system.ch>
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime, time
+
+import pytz
+
+from odoo import api, fields, models
+from odoo.tools import format_date
+
+
+class HrLeaveType(models.Model):
+    _inherit = "hr.leave.type"
+
+    @api.model
+    def get_allocation_data_request(self, target_date=None):
+        """Always include the extra hours leave type in the allocation data."""
+        extra_hours_time_off_type = self.env.ref(
+            "hr_holidays_attendance.holiday_status_extra_hours",
+            raise_if_not_found=False,
+        )
+
+        allocation_data = super().get_allocation_data_request(target_date)
+
+        extra_hours_in_data = any(
+            leave_type[3] == extra_hours_time_off_type.id
+            for leave_type in allocation_data
+        )
+
+        if extra_hours_time_off_type and not extra_hours_in_data:
+            employee = extra_hours_time_off_type.env[
+                "hr.employee"
+            ]._get_contextual_employee()
+            extra_hours_data = extra_hours_time_off_type._prepare_extra_hours_data(
+                employee, target_date
+            )
+            allocation_data.append(extra_hours_data)
+        return allocation_data
```

Comment:
Could you please test that you get the rights `extra_hours_data`  ?

Reply:
Hi @santostelmo and @leemannd, I have added some unit tests.